File: Ontology/ObjectPropertyRestriction.cs
﻿namespace Ontology
{
    public abstract class ObjectPropertyRestriction:
        PropertyRestriction,
        IObjectPropertyRestriction
    {
        protected readonly IObjectPropertyExpression _objectPropertyExpression;

        protected ObjectPropertyRestriction(
            IObjectPropertyExpression objectPropertyExpression
            ) : base(objectPropertyExpression)
        {
            _objectPropertyExpression = objectPropertyExpression;
        }

        IObjectPropertyExpression IObjectPropertyRestriction.ObjectPropertyExpression => _objectPropertyExpression;
    }
}

File: Web/ClientApp/src/app/Ontology/ObjectPropertyRestriction.ts
import { IObjectPropertyRestriction } from "./IObjectPropertyRestriction";
import { IObjectPropertyExpression } from "./IPropertyExpression";
import { PropertyRestriction } from "./PropertyRestriction";

export class ObjectPropertyRestriction
    extends PropertyRestriction
    implements IObjectPropertyRestriction
{
    protected constructor(
        public ObjectPropertyExpression: IObjectPropertyExpression
        )
    {
        super(ObjectPropertyExpression);
    }
}
